app/services/data_migrations/backfill_monthly_statistics_data.rb

```diff
@@ -0,0 +1,13 @@
+module DataMigrations
+  class BackfillMonthlyStatisticsData
+    TIMESTAMP = 20211221113405
+    MANUAL_RUN = false
+
+    def change
+      reports_with_unset_months = Publications::MonthlyStatistics::MonthlyStatisticsReport.where(month: nil)
+      reports_with_unset_months.each do |report|
+        report.update(month: report.created_at.strftime('%Y-%m'))
```

Comment:
This looks like a reasonable assumption but I found out since we last talked about this that there is already at least one report record in prod that was created in the month after it's generation date. So I think we have to work backwards from `created_at` to the previous generation date and use that to work out the month.

Reply:
thanks